extensions/social-asset-cheat-sheet/src/search-sizes.tsx

```diff
@@ -0,0 +1,210 @@
+import { Action, ActionPanel, Detail, Icon, List, LocalStorage, showToast, Toast } from "@raycast/api";
+import { useEffect, useState } from "react";
+import socialSizesData from "./data/socialSizes.json";
+
+interface SocialImageSize {
+  platform: string;
+  type: string;
+  width: number;
+  height: number;
+  icon: Icon;
+  id: string;
+  lastVerified: string;
+  aspectRatio: {
+    simplified: string;
+    decimal: string;
+  };
+}
+
+type PlatformName = "Instagram" | "Twitter/X" | "Facebook" | "LinkedIn" | "Pinterest" | "YouTube";
+
+// Convert JSON data to SocialImageSize objects
+const socialSizes: SocialImageSize[] = socialSizesData.sizes.map((size) => ({
+  ...size,
+  icon: Icon[size.icon as keyof typeof Icon],
+  aspectRatio: calculateAspectRatio(size.width, size.height),
+}));
+
+function calculateAspectRatio(width: number, height: number) {
+  const gcd = (a: number, b: number): number => {
+    while (b) {
+      const temp = b;
+      b = a % b;
+      a = temp;
+    }
+    return a;
+  };
+
+  const divisor = gcd(width, height);
+  const simplifiedWidth = width / divisor;
+  const simplifiedHeight = height / divisor;
+
+  return {
+    simplified: `${simplifiedWidth}:${simplifiedHeight}`,
+    decimal: (width / height).toFixed(2),
+  };
+}
+
+export default function Command() {
+  const [searchText, setSearchText] = useState("");
+
+  // Filter function that handles multiple search terms
+  const filterSizes = (sizes: SocialImageSize[], search: string) => {
+    if (!search.trim()) return sizes;
+
+    const searchTerms = search.toLowerCase().split(/\s+/);
+
+    return sizes.filter((size) => {
+      const searchableText = [
+        size.platform.toLowerCase(),
+        size.type.toLowerCase(),
+        `${size.width}x${size.height}`,
+        size.aspectRatio.simplified,
+        size.aspectRatio.decimal,
+      ].join(" ");
+
+      return searchTerms.every((term) => searchableText.includes(term));
+    });
+  };
+
+  const [favorites, setFavorites] = useState<string[]>([]);
+
+  useEffect(() => {
+    // Load favorites
+    LocalStorage.getItem<string>("favorites").then((savedFavorites) => {
+      if (savedFavorites) {
+        try {
+          setFavorites(JSON.parse(savedFavorites));
+        } catch (error) {
+          showFailureToast(error, { title: "Failed to load favorites" });
+        }
+      }
+    });
+
+    // Check for updates
+    LocalStorage.getItem("lastUpdateCheck").then((lastChecked) => {
+      const now = new Date().toISOString();
+      if (
+        !lastChecked ||
+        (typeof lastChecked === "string" &&
+          new Date(lastChecked).getTime() < new Date().getTime() - 24 * 60 * 60 * 1000)
+      ) {
+        checkForUpdates();
+        LocalStorage.setItem("lastUpdateCheck", now);
+      }
+    });
+  }, []);
+
+  const checkForUpdates = async () => {
+    try {
+      const response = await fetch(
```

Comment:
What about using [useFetch](https://developers.raycast.com/utilities/react-hooks/usefetch) which also caches the results and gives you mapResult/parseResponse and isLoading out of the box 😊